src/lib/y2partitioner/dialogs/bcache.rb

```diff
@@ -66,75 +68,148 @@ def contents
         )
       end
 
-      # Selected caching device. Undefined if result of dialog is not `:next`
+      # Selected caching device
+      #
+      # Undefined if result of dialog is not `:next`.
+      #
       # @return [Y2Storage::BlkDevice, Y2Storage::BcacheCset]
       def caching_device
         @caching.result
       end
 
-      # Selected backing device. Undefined if result of dialog is not `:next`
+      # Selected backing device
+      #
+      # Undefined if result of dialog is not `:next`
+      #
       # @return [Y2Storage::BlkDevice]
       def backing_device
         @backing.result
       end
 
       # Bcache options
+      #
+      # Undefined if result of dialog is not `:next`
+      #
       # @return [Hash<Symbol, Object>]
       def options
         {
           cache_mode: @cache_mode.result
         }
       end
 
-      # Widget to select the backing device
-      class BackingDevice < CWM::ComboBox
-        # @param device [Y2Storage::BlkDevice, nil] existing backing device or nil if it is a new bcache.
-        # @param devices [Array<Y2Storage::BlkDevice>] possible devices that can be used as backing
-        #   device for the Bcache.
-        # @param caching [Y2Partitioner::Dialogs::Bcache::CachingDevice] reference to caching widget.
-        #   Used to check that caching and backing devices are not identical.
-        def initialize(device, devices, caching)
+      # Base class for widgets to select a device
+      #
+      # @see BackingDeviceSelector, CachingDeviceSelector
+      class DeviceSelector < CWM::ComboBox
+        # @return [Y2Storage::Bcache]
+        attr_reader :bcache
+
+        # @return [Array<Y2Storage::BlkDevice, Y2Storage::BcacheCset>]
+        attr_reader :devices
+
+        # Device selected in widget
+        #
+        # Only rely on this value when the dialog succeed and {#store} is called.
```

Comment:
succeeds (third person). Or maybe you meant "succeeded".

Reply:
succeeded